TG-20

evelynda1985 opened the pull request.

Commit:
TG-13  TG-17
Se corrigio el camelCase en el nombre del metodo calcularValorReceta

Commit:
TG-13  TG-17
Se corrigio el camelCase en el nombre del metodo calcularValorReceta

Commit:
TG-13  TG-18
Se anadieron validadores y optimizacion al metodo buscarRecetasPorIngrediente y se probo y funciono :p

Commit:
TG-13  TG-19
Se avanzo en el metodo numeroVerduras, se tiene en este momento un error, y es que esta mostrando el mismo resultado en todas todas las cocinas.

se avanzo en el metodo CalcularReceta, hay un error de parsing...

Commit:
TG-13  TG-18
Se agrego una validacion para que se dijera en el metodo buscarRecetasPorIngrediente que si no encontro en ninguna receta el ingrediente, pues que diga que no se encontro :p

Commit:
TG-13  TG-19
Se corrigio el error de ir sumandole al contador sin importar la cocina. Tambien se esta imprimiento una vez por cocina.

Commit:
TG-13  TG-20
Se corrigieron algunos problemas de cierre de corchetes que me arrojaban mal la informacion.

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -48,13 +49,62 @@ public double valorReceta(Receta[] recetas,String nombreReceta){
+                int cantidadCocina = cocina.getIngredienteCocina()[i].getCantidad();
+                int cantidadReceta = receta.getIngrediente()[j].getCantidad();
+                if (receta.getIngrediente()[i].getNombre().equals(receta.getIngrediente()[j].getNombre())){
+                    if (cocina.getIngredienteCocina()[i].getMedida().equals(receta.getIngrediente()[j].getMedida())){
```
este if no tiene nada en el body por que?

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -48,13 +49,62 @@ public double valorReceta(Receta[] recetas,String nombreReceta){
     
-    public void CalcularValorReceta(Receta[] recetas, String ingrediente){
+    public void buscarRecetasPorIngrediente(Receta[] recetas, String ingrediente){
+        int temp = 0;
```
noppp

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -48,13 +49,62 @@ public double valorReceta(Receta[] recetas,String nombreReceta){
+             if (encontroIngrediente){
+                System.out.println("Puedes usar la receta: " + recetas[i].getNombre() + " con el ingrediente: " + ingrediente);
+                temp = 1;
+             }else if (temp == 0){
```
noppp, mejor un booleano este iff va afuera de los dos for

evelynda1985 replied:
Pero si dejo ese if por fuera, no prodria decirte el nombre de la receta recetas[i].getNombre().

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -48,13 +49,62 @@ public double valorReceta(Receta[] recetas,String nombreReceta){
          }
     }
+    
+    public void numeroVerduras(Cocina[] cocinas){
```
este est bien

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -48,13 +49,62 @@ public double valorReceta(Receta[] recetas,String nombreReceta){
+            int cantidadFaltante = 0;
+            boolean encontroIngrediente = false;
+            for (int j = 0; j < cocina.getIngredienteCocina().length && !encontroIngrediente; j++){
+                int cantidadCocina = cocina.getIngredienteCocina()[i].getCantidad();
```
tienes los indices trucados. En el primer for estas iterando sobre los ingredientes de la receta. Usando
 i < receta.getIngrediente()
Osea que el indice j es el indice para las cocinas.
De casualidad hay dos cocinas y dos recetas entonces te fuciona. Ten cuidado con los indices. 

Corregir todos los indices.

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -48,13 +49,62 @@ public double valorReceta(Receta[] recetas,String nombreReceta){
+                    if (cocina.getIngredienteCocina()[i].getMedida().equals(receta.getIngrediente()[j].getMedida())){
+                    }
+                    if (cantidadCocina >= cantidadReceta){
+                            System.out.println("Hay Sueficiente Ingrediente " + receta.getIngrediente()[j].getNombre());
```
me imagino q este bloque va si la medidas son iguales.

Commit:
TG-13  TG-20
Se corrigieron la posicion de los if en los metodos buscarRecetasPorIngrediente y calcularReceta(), tambien se corrigio el for de calcularReceta(). Pendiente revisar el metodo buscarRecetasPorIngrediente.

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -50,18 +50,16 @@ public double valorReceta(Receta[] recetas,String nombreReceta){
                 System.out.println("Puedes usar la receta: " + recetas[i].getNombre() + " con el ingrediente: " + ingrediente);
-                temp = 1;
-             }else if (temp == 0){
+             }else{
```
esto no sigue sirviendo lo probaste?

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -81,22 +79,22 @@ public void numeroVerduras(Cocina[] cocinas){
-                    if (cocina.getIngredienteCocina()[i].getMedida().equals(receta.getIngrediente()[j].getMedida())){
-                    }
+                    if (cocina.getIngredienteCocina()[i].getMedida().equals(receta.getIngrediente()[j].getMedida())){                   
                     if (cantidadCocina >= cantidadReceta){
```
trata de identar

jramirezl replied:
los if!

Commit:
TG-13  TG-20
Se agregaron algunos else a los metodos de la clase Chef, se hizo prueba de escritorio, se evaluaron diferentes escenarios. Espero que esta bien si haya quedado bien :p

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -25,44 +25,65 @@ public void cocinarReceta(Cocina objCocina, Receta objReceta)
         Ingrediente [] ingredientesCocina = objCocina.getIngrediente();
-        
+        boolean hayIngrediente = false;
+        boolean hayIngredienteCocina = false;
```
estas usando los dos booleanso pa lo mismo

jramirezl replied:
debes probar este codigo
